src/test/store/receive-profile.test.js

```diff
@@ -368,6 +371,124 @@ describe('actions/receive-profile', function() {
+      const browsingContextID = 123;
+      const innerWindowID = 111111;
+      function setup(profile: ?Profile) {
+        const store = blankStore();
+
+        if (!profile) {
+          profile = getEmptyProfile();
+          profile.threads.push(
+            // This thread should be completely hidden because it doesn't contain anything from the tab.
+            getEmptyThread({
+              name: 'GeckoMain',
+              processType: 'default',
+              pid: 1,
+            }),
+            // This thread shouldn't be hidden because it will have markers with innerWindowID.
+            getEmptyThread({ name: 'GeckoMain', processType: 'tab', pid: 2 }),
+            // This thread should be completely hidden because it doesn't contain anything from the tab.
+            getEmptyThread({ name: 'GeckoMain', processType: 'tab', pid: 3 })
+          );
+        } else {
+          // Appending a thread to test the second all the time
+          profile.threads = [
+            getEmptyThread({
+              name: 'GeckoMain',
+              processType: 'default',
+              pid: 1,
+            }),
+            ...profile.threads,
+          ];
+        }
+
+        profile.meta.configuration = {
+          threads: [],
+          features: [],
+          capacity: 1000000,
+          activeBrowsingContextID: browsingContextID,
+        };
+        profile.pages = [
+          {
+            browsingContextID: browsingContextID,
+            innerWindowID: innerWindowID,
+            url: 'URL',
+            embedderInnerWindowID: 0,
+          },
+        ];
+
+        addMarkersToThreadWithCorrespondingSamples(profile.threads[1], [
+          [
+            'RefreshDriverTick',
+            0,
+            {
+              type: 'tracing',
+              category: 'Navigation',
+              interval: 'start',
+              innerWindowID: innerWindowID,
+            },
+          ],
+          ...getNetworkMarkers({
+            startTime: 1,
+          }),
+        ]);
+
+        store.dispatch(viewProfile(profile));
+        store.dispatch(changeShowTabOnly(browsingContextID));
+
+        return { ...store, profile };
+      }
+
+      it('should calculate the hidden tracks correctly', function() {
+        const { getState } = setup();
+        expect(getHumanReadableTracks(getState())).toEqual([
+          'show [thread GeckoMain tab] SELECTED',
+        ]);
+      });
+
+      it('should not hide screenshot tracks', function() {
+        const profile = getScreenshotTrackProfile();
+        profile.threads[0].name = 'GeckoMain';
+        profile.threads[0].processType = 'tab';
+
+        const { getState } = setup(profile);
+        expect(getHumanReadableTracks(getState())).toEqual([
+          'show [screenshots]',
+          'show [screenshots]',
+          'show [thread GeckoMain tab] SELECTED',
+        ]);
+      });
+
+      it('should calculate the hidden local threads correctly', function() {
```

Comment:
These tests looks good to me!

We should also probably try out some tests for when things go badly, and all tracks are hidden. I left some comments in the implementation around some of my concerns around this.

Reply:
#2438